content/blogposts/2023/cody-vscode-0-10-release.md

```diff
@@ -0,0 +1,105 @@
+
+### Chat history JSON export
+
+The new "Export" button in Chat History allows you to export and inspect your chat history as JSON:
+
+<img alt="Screenshot of Cody v0.10 chat JSON export buttton" src="https://storage.googleapis.com/sourcegraph-assets/cody-vscode-0.10.0-chat-export.png" style={{marginTop:"1rem",marginBottom:"1.5rem",marginLeft:0,maxWidth:306}} />
+
+The export includes the full conversation between Cody and the LLM, allowing you to see how Cody works, send your chat history to Sourcegraph support, and more easily develop custom commands.
+
+```json
+{
+    "Sun, 03 Sep 2023 14:00:44 GMT": {
+        "id": "2023-09-03T14:00:44.696Z",
+        "interactions": [
+            {
+                "humanMessage": { ... },
+                "assistantMessage": { ... },
+                "fullContext": [
+                    { ... }
+                ]
+            }
+        ]
+        
+    }
+}
+```
+
+### Updated custom command JSON format
+
+Custom Cody Commands is an experimental feature for extending and customizing Cody. They can be configured locally, or shared with your team by checking them in to your repository.
+
+We’ve updated the JSON configuration format so it now requires a `slashCommand` (it's used as the configuration key) and replaced the command name with a sentence-cased `description`:
+
+```diff
+{
+    "commands": {
+-       "My Custom Command": {
++       "my-custom-command": {
+-           "slashCommand": "/my-custom-command",
++           "description": "My custom command",
+            "command": "node my-custom-context.js",
+            "prompt": "My custom prompt"
+        }
+    }
+}
+```
+
+Old configuration files will be automatically updated to the new format. It's recommended you also update your custom command descriptions to use sentence case, to match the built-in Cody commands.
+
+### Autocomplete rate limit visibiity
+
+We’ve improved the visiblity of [autocompletion rate limits](https://docs.sourcegraph.com/cody/troubleshooting#autocomplete-rate-limits). When a rate limit is being applied the Cody statusbar icon now changes color. The settings menu also provides information about the rate limit, and links through to the documentation.
+
+<img alt="Screenshot of Cody v0.10 showing an autocomplete warning" src="https://storage.googleapis.com/sourcegraph-assets/cody-vscode-0.10.0-autocomplete-rate-limit.png" style={{marginTop:"1rem",marginBottom:"1.5rem",marginLeft:0,maxWidth:613}} />
+
+### Faster autocomplete with TLS connection reuse
+
+We added support for reusing TLS connections, resulting in a 5-10% speed up of autocomplete responses. Due to VS Code limitations, to enable TLS connection reuse you must configure your VS Code "Http: Proxy Support" setting to "off":
+
+```json
+{
+    "http.proxySupport": "off"
+}
+```
```

Comment:
Great… makes sense!

It's always important to mention what exactly changed though, not just the result. How's this sound @philipp-spiess?

> Autocompletions now accept a streamed response from the server, and terminate the stream processing as soon as possible instead of waiting for the full response. These two changes have resulted in a 40% speed up of multi-line completions, and 10% speedup for single-line completions.

Reply:
that sounds much better, thanks for the feedback! slowly but surely getting better at public writing